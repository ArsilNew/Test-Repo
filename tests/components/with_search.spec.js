import React from "react";
import renderer from "react-test-renderer";

import withSearch from "../../src/components/with_search.js";
import SourceList from "../../src/components/source_list";

const ListComponent = withSearch(SourceList);
const showSearch = true;
const messages = { searchPlaceholder: "Search" };

describe("With search", () => {
  test("renderer search", () => {
    const component = renderer.create(
      <ListComponent showSearch={showSearch} messages={messages} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("renderer user search", () => {
    const UserSearch = props => <input type="text" {...props} />;

    const component = renderer.create(
      <ListComponent
        searchRenderer={UserSearch}
        showSearch={showSearch}
        messages={messages}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
