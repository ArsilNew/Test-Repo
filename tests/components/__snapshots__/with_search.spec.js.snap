// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With search Case custom search component 1`] = `
<CustomComponent
  filteredItems={
    Array [
      Object {
        "id": 11,
        "label": "item-11",
      },
    ]
  }
>
  <UserSearch
    onChange={[MockFunction]}
    searchIcon={
      <img
        alt=""
        src="custom_icon"
      />
    }
    searchPlaceholder="Search"
    value="11"
  />
</CustomComponent>
`;

exports[`With search Case default search component 1`] = `
<CustomComponent
  filteredItems={
    Array [
      Object {
        "id": 1,
        "label": "item-1",
      },
    ]
  }
>
  <Search
    onChange={[MockFunction]}
    searchIcon={
      <img
        alt=""
        src="custom_icon"
      />
    }
    searchPlaceholder="Search"
    value="1"
  />
</CustomComponent>
`;

exports[`With search Case not search component 1`] = `<CustomComponent />`;
